akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala

```diff
@@ -201,43 +202,56 @@ object ShardedEntity {
+   * Any [[Behavior]] can be used as a sharded entity actor, but the combination of sharding and persistent actors
+   * is very common and therefore the [[ShardedEntity.ofPersistentEntity]] is provided as convenience.
+   *
    * @param typeKey A key that uniquely identifies the type of entity in this cluster
+   * @param createBehavior Create the behavior for an entity given an entityId and more in the [[ShardedEntityContext]]
    * @param stopMessage Message sent to an entity to tell it to stop, e.g. when rebalanced or passivated.
-   *
    * @tparam M The type of message the entity accepts
    */
-  def create[M](
-    createBehavior: JFunction[String, Behavior[M]],
+  def of[M](
     typeKey:        EntityTypeKey[M],
+    createBehavior: JFunction[ShardedEntityContext[M], Behavior[M]],
     stopMessage:    M): ShardedEntity[M, ShardingEnvelope[M]] = {
-    create(new BiFunction[ActorRef[ClusterSharding.ShardCommand], String, Behavior[M]] {
-      override def apply(shard: ActorRef[ClusterSharding.ShardCommand], entityId: String): Behavior[M] =
-        createBehavior.apply(entityId)
-    }, typeKey, stopMessage)
+    new ShardedEntity(createBehavior, typeKey, stopMessage, Props.empty, Optional.empty(), Optional.empty(), Optional.empty())
   }
 
   /**
-   * Defines how the entity should be created. Used in [[ClusterSharding#start]]. More optional
-   * settings can be defined using the `with` methods of the returned [[ShardedEntity]].
+   * Defines how the [[PersistentEntity]] should be created. Used in [[ClusterSharding#start]]. Any [[Behavior]] can
+   * be used as a sharded entity actor, but the combination of sharding and persistent actors is very common
+   * and therefore this factory is provided as convenience.
+   *
+   * More optional settings can be defined using the `with` methods of the returned [[ShardedEntity]].
    *
-   * @param createBehavior Create the behavior for an entity given `ShardCommand` ref and an entityId
    * @param typeKey A key that uniquely identifies the type of entity in this cluster
+   * @param createPersistentEntity Create the `PersistentEntity` for an entity given an entityId and more in the [[ShardedEntityContext]]
    * @param stopMessage Message sent to an entity to tell it to stop, e.g. when rebalanced or passivated.
-   * @tparam M The type of message the entity accepts
+   * @tparam Command The type of message the entity accepts
    */
-  def create[M](
-    createBehavior: BiFunction[ActorRef[ClusterSharding.ShardCommand], String, Behavior[M]],
-    typeKey:        EntityTypeKey[M],
-    stopMessage:    M): ShardedEntity[M, ShardingEnvelope[M]] =
-    new ShardedEntity(createBehavior, typeKey, stopMessage, Props.empty, Optional.empty(), Optional.empty(), Optional.empty())
+  def ofPersistentEntity[Command, Event, State >: Null](
+    typeKey:                EntityTypeKey[Command],
+    createPersistentEntity: JFunction[ShardedEntityContext[Command], PersistentEntity[Command, Event, State]],
```

Comment:
Should we avoid type rename/aliases like this in public APIs or do they look alright from Java?


Also the null bound is weird, but I think I saw another issue mentioning looking into that?

Reply:
> Should we avoid type rename/aliases like this in public APIs or do they look alright from Java?

IntelliJ show it as `Function` and javadoc looks good 
https://github.com/akka/akka/pull/25697#discussion_r224118919
https://doc.akka.io/japi/akka/current/akka/actor/typed/javadsl/ActorContext.html#messageAdapter-java.lang.Class-java.util.function.Function-

> Also the null bound is weird, but I think I saw another issue mentioning looking into that?

yes, that is tracked elsewhere